File: src/main/java/io/gluth/pagespace/domain/Page.java
package io.gluth.pagespace.domain;

import java.util.Objects;

public final class Page {

    private final String id;
    private final String title;

    public Page(String id, String title) {
        this.id = Objects.requireNonNull(id, "id");
        this.title = Objects.requireNonNull(title, "title");
    }

    public String id() { return id; }
    public String title() { return title; }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Page other)) return false;
        return id.equals(other.id);
    }

    @Override
    public int hashCode() { return id.hashCode(); }

    @Override
    public String toString() { return "Page[" + id + ", " + title + "]"; }
}

File: src/main/kotlin/io/gluth/pagespace/domain/Page.kt
package io.gluth.pagespace.domain

class Page(val id: String, val title: String) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Page) return false
        return id == other.id
    }

    override fun hashCode(): Int = id.hashCode()

    override fun toString(): String = "Page[$id, $title]"
}
